### PR TITLE
Adds option to change components path

### PR DESCRIPTION
**Pull request by lfalcao**
Related: #863 :clinking_glasses:

### CODE REVIEW

File: docs/guide/generators.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ bin/rails generate component Example title content
 
 ## Options
 
+ViewComponent uses `app/components` as folder for components, but you can change it by adding `config.view_component.view_component_path` into your applicaton config.
```
Comment (Spone) [inline docs/guide/generators.md:25]: The markdown linter is complaining about a missing empty line.

```suggestion
ViewComponent uses `app/components` as folder for components, but you can change it by adding `config.view_component.view_component_path` into your applicaton config.

```

File: lib/rails/generators/component/component_generator.rb [hunk 2]
```diff
@@ -44,6 +44,10 @@ def initialize_body
       def initialize_call_method_for_inline?
         options["inline"]
       end
+
+      def component_path
+        Rails.application.config.view_component.view_component_path || 'app/components'
```
Comment (Spone) [inline lib/rails/generators/component/component_generator.rb:49]:
```suggestion
        Rails.application.config.view_component.view_component_path || "app/components"
```

File: lib/rails/generators/erb/component_generator.rb [hunk 3]
```diff
@@ -19,15 +19,19 @@ def copy_view_file
...
+          File.join(component_path, class_path, "#{file_name}_component.html.erb")
         end
       end
 
       def file_name
         @_file_name ||= super.sub(/_component\z/i, "")
       end
+
+      def component_path
+        Rails.application.config.view_component.view_component_path || 'app/components'
```
Comment (Spone) [inline lib/rails/generators/erb/component_generator.rb:33]:
```suggestion
        Rails.application.config.view_component.view_component_path || "app/components"
```

File: lib/rails/generators/haml/component_generator.rb [hunk 4]
```diff
@@ -18,15 +18,19 @@ def copy_view_file
...
+          File.join(component_path, class_path, "#{file_name}_component.html.haml")
         end
       end
 
       def file_name
         @_file_name ||= super.sub(/_component\z/i, "")
       end
+
+      def component_path
+        Rails.application.config.view_component.view_component_path || 'app/components'
```
Comment (Spone) [inline lib/rails/generators/haml/component_generator.rb:32]:
```suggestion
        Rails.application.config.view_component.view_component_path || "app/components"
```

File: lib/rails/generators/slim/component_generator.rb [hunk 5]
```diff
@@ -18,15 +18,19 @@ def copy_view_file
...
+          File.join(component_path, class_path, "#{file_name}_component.html.slim")
         end
       end
 
       def file_name
         @_file_name ||= super.sub(/_component\z/i, "")
       end
+
+      def component_path
+        Rails.application.config.view_component.view_component_path || 'app/components'
```
Comment (Spone) [inline lib/rails/generators/slim/component_generator.rb:32]:
```suggestion
        Rails.application.config.view_component.view_component_path || "app/components"
```

File: test/generators/component_generator_test.rb [hunk 6]
```diff
@@ -126,4 +126,13 @@ def test_invoking_haml_template_engine
...
+
+  def test_generating_components_with_custom_component_path
+    with_custom_component_path("app/parts") do
+      run_generator %w[user]
+
+      assert_file "app/parts/user_component.rb"
+      assert_file "app/parts/user_component.html.erb"
+    end
+  end
+end
```
Comment (Spone) [inline test/generators/component_generator_test.rb:138]:
```suggestion
end

```

File: test/test_helper.rb [hunk 7]
```diff
@@ -106,4 +113,4 @@ def with_render_monkey_patch_config(enabled)
   yield
 ensure
   ViewComponent::Base.render_monkey_patch_enabled = old_default
-end
+end
```
Comment (Spone) [inline test/test_helper.rb:116]:
```suggestion
end

```

**Comment by Spone** [comment]
Looking good, thanks for contributing!

I'm wondering if we should try to make it DRY by moving the `component_path` method declaration to a `Utils` module (and including it in generator classes).

**Comment by lfalcao** [comment]
@Spone Done, I thought in do that but in another time... :blush:

File: docs/guide/generators.md [hunk 8]
```diff
@@ -22,6 +22,8 @@ bin/rails generate component Example title content
...
```
Comment (joelhawksley) [inline docs/guide/generators.md:25]:
```suggestion
Generated ViewComponents are added to `app/components` by default. Set `config.view_component.view_component_path` to use a different path.
```

**Review by joelhawksley — APPROVED** [review]

**Comment by joelhawksley** [comment]
@Spone this looks good to me. Feel free to merge when you think it's ready.

**Review by Spone — APPROVED** [review]

**Review by Spone — CHANGES_REQUESTED** [review]
@lfalcao I just noticed this PR is missing a changelog entry, can you add one please?

**Comment by Spone** [comment]
Thanks @lfalcao, can you please rebase and solve the changelog conflict? Then I'll go ahead and merge!

**Comment by lfalcao** [comment]
@Spone ouch, just a moment.

**Comment by lfalcao** [comment]
@Spone :bow:

**Review by Spone — APPROVED** [review]

**Pull request merged**